File: solutions/java/log-levels/2/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        int index = logLine.indexOf(" ");
        String msg = logLine.substring(index + 1);
        String trimmedString = msg.trim();
        return trimmedString;
    }

    public static String logLevel(String logLine) {
        String result = logLine.replaceAll("\\s", "");
        int index = logLine.indexOf("]");
        String msg = logLine.substring(1, index).toLowerCase();
        return msg;
    }

    public static String reformat(String logLine) {
        String msg = message(logLine);
        String log = logLevel(logLine);

        String finalMsg = msg + " (" + log + ")";
        return finalMsg;
    }
}

File: solutions/csharp/log-levels/2/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int index = logLine.IndexOf(" ");
        string msg = logLine.Substring(index + 1);
        string trimmed = msg.Trim();
        return trimmed;
    }

    public static string LogLevel(string logLine)
    {
        int index = logLine.IndexOf("[");
        int lastIndex = logLine.IndexOf("]");
        string msg = logLine.Substring(index + 1, lastIndex - 1);
        string trimmedLower = msg.Trim().ToLower();
        return trimmedLower;
    }

    public static string Reformat(string logLine)
    {
        string msg = Message(logLine);
        string log = LogLevel(logLine);
        return $"{msg} ({log})";
    }
}
